src/main/java/uk/gov/pay/ledger/transaction/search/common/TransactionSearchParamsValidator.java

```diff
@@ -6,22 +6,21 @@
 import java.time.ZonedDateTime;
 import java.time.format.DateTimeParseException;
 
-import static org.apache.commons.lang3.StringUtils.isBlank;
+import static com.google.common.collect.Iterables.isEmpty;
 import static org.apache.commons.lang3.StringUtils.isNotBlank;
 
 public class TransactionSearchParamsValidator {
     private static final String FROM_DATE_FIELD = "from_date";
     private static final String TO_DATE_FIELD = "to_date";
 
-    public static void validateSearchParams(TransactionSearchParams searchParams, String gatewayAccountId) {
+    public static void validateSearchParams(TransactionSearchParams searchParams, CommaDelimitedSetParameter gatewayAccountIds) {
         if (isNotBlank(searchParams.getFromDate())) {
             validateDate(FROM_DATE_FIELD, searchParams.getFromDate());
         }
         if (isNotBlank(searchParams.getToDate())) {
             validateDate(TO_DATE_FIELD, searchParams.getToDate());
         }
-
-        if(searchParams.getWithParentTransaction() && isBlank(gatewayAccountId)){
+        if(searchParams.getWithParentTransaction() && isEmpty(gatewayAccountIds.getParameters())){
```

Comment:
You can just do `gatewayAccountIds.getParameters().isEmpty()` and avoid the need for Guava.

Reply:
not if it's null